1-Draft/RFCNNNN-Download-Cmdlet.md

```diff
@@ -0,0 +1,294 @@
+
+There is a gap in functionality between PowerShell and other tools and shells.
+This gap was made apparent by the inclusion of `curl.exe` in Windows.
+While PowerShell and the Web Cmdlets should not seek 100% parity with similar technologies,
+The ability to download files over multiple protocols is an essential feature.
+A PowerShell user will want a PowerShell native way to perform this task.
+
+As there is a feature gap that is essential to fill
+and the current Web Cmdlets are not an ideal candidate for closing that gap,
+a new cmdlet should be added whose primary objective is to support file downloads over multiple protocols.
+
+## Specification
+
+`Invoke-Download` shall be the name of the cmdlet with the alias of `idl`.
+
+### Syntax
+
+The syntax shall be as follows:
+
+```none
+Invoke-Download -Uri <Uri> [-Credential <PSCredential>]
+    [-Path <String> | -LiteralPath <String>]
+    [-Proxy <Uri>] [-ProxyCredential <PSCredential>]
+    [-Resume] [-PassThru] [-AsJob]
+```
+
+```none
+idl -Uri <Uri> [-Credential <PSCredential>]
+    [-Path <String> | -LiteralPath <String>]
+    [-Proxy <Uri>] [-ProxyCredential <PSCredential>]
+    [-Resume] [-PassThru] [-AsJob]
+```
+
+Examples:
+
+```powershell
+idl https://i.imgur.com/gXnRf4J.jpg
+Invoke-Download -Uri 'https://i.imgur.com/gXnRf4J.jpg'
+Invoke-Download -Uri 'https://i.imgur.com/gXnRf4J.jpg' -Path 'c:\images\rin.jpg'
+Invoke-Download -Uri 'https://i.imgur.com/gXnRf4J.jpg' -PassThru | Get-content
+```
+
+### Relationship with Web Cmdlets
+
+The cmdlet shall not share base code with the existing Web Cmdlets.
+The cmdlet shall have a subset of the Web Cmdlets features that are specific or essential to downloading files.
+For example, the cmdlet will not support body content nor HTTP methods other than GET.
+
+As the cmdlet is not centered around HTTP and HTTPS,
+the available features need to be generalized for multiple protocols.
+To this end, advanced authentication features in the Web Cmdlets will not exist in the download cmdlet.
+Most protocols expect a username and password.
+At minimum, a `PSCredential` will be accepted.
+
+### Automatic Filename Resolution Feature
+
+A key feature of the cmdlet is to require nothing more than a URI.
+Commands such as `curl` and `wget` have long supported automatic file name resolution.
+This allows the user to supply just a URI and the file will be saved in the current working directory.
+The file name is automatically chosen based on the URI and the presence of existing files.
+
+The cmdlet will automatically chose a file name when a path is not supplied.
+The filename will be chosen in this manner:
+
+1. The last segment of `Uri.Segments`
+1. In the case that the last segment is `/`, the file name `index.html` will be used
+1. Illegal file name characters will be replaced with their URL encoded equivalent.
+1. If a file with that name is present, the basename will be appended with a period followed by `DateTime.Now.ToString("yyyMMddHHmmssfff")`
```

Comment:
Is there a reason you decided not to follow the convention of adding `(n)` to the filename?  The file itself will have a creation timestamp.